### ISSUE TITLE
run-script is unable to find zef in certain circumstances

### ISSUE THREAD

**Issue by JJ** [rakudo/rakudo#4674]
I'm trying to set up a Docker image that contains zef, and it works locally, it does not work in a CI environment such as GitHub. This is essentially the same as [this](https://github.com/ugexe/zef/issues/430), since the error occurs here.

## Context

Created a series of images that use Alpine as base. Since GitHub Actions use UID 1001 for their user, I've created a container that has a `raku` user with that UID. zef works perfectly... locally. However, [this happens](https://github.com/JJ/p6-pod-load/runs/4573911222?check_suite_focus=true) in GitHub actions: zef is found, however zef is unable to find its own library module yielding a 

```
===SORRY!===
No candidate found for 'zef'.
```

## Expected Behavior

Work the same as locally

## Actual Behavior

No zef command works, although it's in the path and it's actually run.

## Steps to Reproduce

Check out [failing GitHub actions here](https://github.com/JJ/p6-pod-load)

## Your Environment

* raku -v

2021.10

* zef list --installed
```
===> Found via /usr/share/perl6/core
CORE:ver<6.d>:auth<perl>
===> Found via /home/raku/.raku
File::Find:ver<0.1.1>
File::Which:ver<1.0.1>
LibraryMake:ver<1.0.0>:auth<github:retupmoca>
Linenoise:ver<0.1.1>:auth<Rob Hoelz>
Shell::Command
zef:ver<0.13.4>:auth<github:ugexe>:api<0>
```

**Comment by JJ** [comment]
OK, I see what's the deal here. The repo-chain shows this:

```
(inst#/github/home/.raku inst#/usr/share/perl6/site inst#/usr/share/perl6/vendor inst#/usr/share/perl6/core ap# nqp# perl5#)
```

While locally it's simply /home/raku. Not really clear why that happens.

**Comment by jonathanstowe** [comment]
Yeah I've seen similar when building images for Elastic Beanstalk using Jenkins, I don't think it's specific to GH actions.

**Comment by JJ** [comment]
The problem is that Rakudo is not keeping track of where the repositories are, but using sensible defaults, in this case, apparently, reading the `$HOME` environment variable. If the value of that variable while installing something is different to the value of the variable now, there's trouble.

**Comment by lizmat** [comment]
Is this still an issue?
